### PR TITLE
Expose pod IP as jaeger process tag in go services

### PR DESCRIPTION
**Pull request by dmitryax**

### CODE REVIEW

File: src/checkoutservice/main.go [hunk 1]
```diff
@@ -131,6 +132,9 @@ func initJaegerTracing() {
 		Endpoint: fmt.Sprintf("http://%s", svcAddr),
 		Process: jaeger.Process{
 			ServiceName: "checkoutservice",
+			Tags: []jaeger.Tag{
+				jaeger.StringTag(k8sIPLabelName, os.Getenv("POD_IP")),
```
Comment (tigrannajaryan) [inline src/checkoutservice/main.go:136]: Based on how @owais described k8s functionality this should not be needed.

**Comment by tigrannajaryan** [comment]
@dmitryax Let's not do this for now and wait for @owais 

(Can you add Owais to this repo too?)

**Comment by tigrannajaryan** [comment]
Nevermind, I added him.

**Comment by owais** [comment]
Tigran is right, this isn't required. k8s.pod.ip is detected and controlled by the Otel collector/agent. Some clients do try to detect their remote IP and ship that as a tag but they use `ip` key for that and k8s processor respects that as well. Jaeger's Go client has code to do it as well.

https://github.com/jaegertracing/jaeger-client-go/blob/8e7ec7b577ab1446d96a02320b2fb092eca7a66f/tracer.go#L159-L161
https://github.com/jaegertracing/jaeger-client-go/blob/f7e0d4744fa6d5287c53b8ac8d4f83089ce07ce8/utils/localip.go#L54

Did spans generated with jaeger-client-go not have the `ip` tag for your service?

**Comment by dmitryax** [comment]
these services use an old opensensus instrumentation. let me try jaeger then

**Comment by owais** [comment]
OK, that explains it. OC does not do it AFAIK. So you are using OC to instrument but export in Jaeger thrift? Any reason to not export in OC if you are using OC instrumentation? Or Zipking or Jaeger gRPC?

**Comment by dmitryax** [comment]
That's how it was instrumented already. @tigrannajaryan should we instrument it with OTLP instead?

**Comment by tigrannajaryan** [comment]
Let's not use OTLP yet, we are discussing some breaking changes to it. Can we use JaegerGRPC or Zipkin? 

@owais Alternatively is there a way to fix pod ip population in Collector for Jaeger thrift?

**Comment by owais** [comment]
If this is just for demo, I'd suggest to use zipkin. All instrumentations support zipkin as far as I know and it's known to work well. In the meanwhile I'll try to add support for jaeger thrift.

**Comment by tigrannajaryan** [comment]
> If this is just for demo, I'd suggest to use zipkin. All instrumentations support zipkin as far as I know and it's known to work well. In the meanwhile I'll try to add support for jaeger thrift.

@dmitryax can we go with Zipkin for the demo? BTW, we can enable several receivers in Collector config, it should be OK to enable to all open-source ones by default.

**Comment by dmitryax** [comment]
Sounds good. I can switch it to zipkin. Should we prefer zipkin for other demo app services as well?

**Comment by tigrannajaryan** [comment]
> Sounds good. I can switch it to zipkin. Should we prefer zipkin for other demo app services as well?

Zipkin seems to be the safe bet from functional perspective for now. It is probably not the best for performance tests though since it is the slowest protocol in Collector. For now it is OK.

**Comment by owais** [comment]
@tigrannajaryan @dmitryax PR to support this for Jaeger Thrift HTTP receiver. https://github.com/open-telemetry/opentelemetry-collector/pull/938

**Comment by dmitryax** [comment]
Thanks Owais!

**Comment by dmitryax** [comment]
@tigrannajaryan @owais need to update core dep in contrib to use this Jaeger Thrift HTTP receiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/224 PTAL

**Comment by tigrannajaryan** [comment]
@dmitryax is this PR still needed?

**Comment by dmitryax** [comment]
Yes, let me update it

**Comment by dmitryax** [comment]
should not be needed anymore. let me test if jaeger receiver does the trick now

**Comment by dmitryax** [comment]
it works now. Thanks @owais ! Closing this PR

**Pull request closed without merging**